lib/LaTeXML/Common/Error.pm

```diff
@@ -50,6 +72,127 @@ sub colorizeString {
   return ($COLORIZED_LOGGING && $color_scheme{$alias}
     ? colored($string, $color_scheme{$alias})
     : $string); }
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Log file
+# Initially: possibility of a single log file, as well as STDERR
+# General idea: everything goes to the log file (if any)
+# only a little bit goes to STDERR (controllable by verbosity)
+# Is there a single DWIM strategy that puts the right things on each stream?
+our $LOG;
+our $LOG_PATH;
+# NOTE: since LaTeXML.pm (currently) repeatedly opens & closes the log,
+# and doesn't (YET) know whether there's already a log open,
+# this bit of hackery only keeps the outermost log open. Fix this!
+our $log_count = 0;
+# Where? Current directory? (probably) Source directory? (probably not)
+# Option for appending?
+# Note that the $path can be a reference to a string (which gets appended to)
+sub OpenLog {
+  my ($path, $append) = @_;
+  $log_count++;
+  return if $LOG or !$path;    # already opened?
+  open($LOG, ($append ? '>>' : '>'), $path) or die "Cannot open log file $path for writing: $!";
+  $LOG_PATH = $path;
+  binmode($LOG, ":encoding(UTF-8)");
+  return; }
+
+# Should be sure it autocloses
+sub CloseLog {
+  $log_count--;
+  return if !$LOG || $log_count;
+  close($LOG) or die "Cannot close log file: $!";
+  $LOG = undef;
+  NoteStatus("Please see $LOG_PATH for details")
+    if $IS_TERMINAL;    # TEMPORARY HACK until LaTeXML.pm is sorted.
+  return; }
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Low-level I/O
+
+# print one (or more) lines to the Log and STDERR, according to $loglevel, $termlevel, respectively.
+# If verbosity == $termlevel, only prints the 1st line of the message.
+# Print to STDERR if verbosity is >= $level; only first line if verbosity is >= $shortlevel.
+# Starts a fresh line by pushing any Spinner line ahead.
+# ORRR should $level also apply to Logfile ???
+sub _printline {
+  my ($loglevel, $termlevel, $message) = @_;
+  return if ($VERBOSITY < $loglevel) && ($VERBOSITY < $termlevel);
+  $message =~ s/^\n+//s;    # Strip newlines off ends.
+  $message =~ s/\n+$//s;
+  my $clean_message = ($LOG || !$IS_TERMINAL ? strip_ansi($message) : $message);
+  $message = $clean_message unless $IS_TERMINAL;
+  print $LOG _freshline($LOG), $clean_message, "\n" if $LOG && ($VERBOSITY >= $loglevel);
+
+  _spinnerclear();
+  if ($VERBOSITY > $termlevel) {
+    print STDERR _freshline(\*STDERR), $message, "\n"; }
```

Comment:
I did some first testing with the web service behind latexml.mathweb.org/ today, and using this particular bit of logic is not quite backwards compatible as it generates a bunch of STDERR chatter in the same flow as the web service HTTP logging: 

```
[2021-05-14 16:14:08.81069] [52595] [debug] [5p5kb_szNuWZ] POST "/convert"
[2021-05-14 16:14:08.81105] [52595] [debug] [5p5kb_szNuWZ] Routing to a callback
Please see SCALAR(0x55aa693784d8) for details
Please see SCALAR(0x55aa693784d8) for details
Please see SCALAR(0x55aa693784d8) for details
ltxmojo (LaTeXML version 0.8.5; revision 57ad6343)
Conversion complete: No obvious problems
Post-processing complete: No obvious problems
Status:conversion:0
Please see SCALAR(0x55aa693784d8) for details
[2021-05-14 16:14:09.93501] [52595] [debug] [5p5kb_szNuWZ] 200 OK (1.1243s, 0.889/s)
```

The `print $LOG` a few lines above here is everything that the web service expects, and I'd claim everything LaTeXML.pm should offer by default.

The then following `print STDERR` bits ought to be explicitly requested by the executable, if we are to remain backwards compatible with LaTeXML.pm. Maybe an extra parameter to `OpenLog`, which makes it clear if `STDERR` should be enabled or suppressed. It is quite nice that we fully control the callers to `OpenLog`, so as long as we configure it correctly the third-party applications may never have to find out something changed in the internal API for logging.

While some cases (such as latexmls), would automatically resolve this by redirecting their STDERR to `/dev/null` before calling `convert()`, others wouldn't - a Mojolicious web service uses STDERR by default in debug mode, or when deployed via morbo, as one example - but currently is expecting all of latexml's messaging to be supplied via the `$response{log}` object, without any STDERR use.